gflownet/envs/base.py

```diff
@@ -411,18 +411,60 @@ def step_backwards(
         self.n_actions += 1
         return self.state, action, True
 
-    def sample_actions(
+    # TODO: do not apply temperature here but before calling this method.
+    # TODO: rethink whether sampling_method should be here.
+    def sample_actions_batch(
         self,
         policy_outputs: TensorType["n_states", "policy_output_dim"],
-        sampling_method: str = "policy",
-        mask_invalid_actions: TensorType["n_states", "policy_output_dim"] = None,
-        temperature_logits: float = 1.0,
-        max_sampling_attempts: int = 10,
+        mask: Optional[TensorType["n_states", "policy_output_dim"]] = None,
+        states_from: Optional[List] = None,
+        is_backward: Optional[bool] = False,
```

Comment:
I assume this is a known issue since you asked on Slack, but the mix of `is_backward`/`is_forward` that's in the code right now is a little confusing.

Reply:
I'm aware and I agree, that's why I asked, but I would call this out of the scope of this rather urgent PR.

Reply:
Oh, absolutely; maybe worth creating an issue.